### PR TITLE
Stopping the currently running test

### PR DESCRIPTION
**Pull request by kevgo**
implements #55 

@charlierudolph

### CODE REVIEW

File: features/killing-runs.feature [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    Then the process is still running
+
+
+  Scenario: no previous test
+    Given Tertestrial is running
+    When sending the command:
+      """
+      {"stopCurrentTest": true}
+      """
+    Then the process is still running
```
Comment (charlierudolph) [inline features/killing-runs.feature]: What gets printed here? I think if there is nothing running we should print an error message

**Comment by kevgo** [comment]
Added error messages. Ready for another review!

File: features/stopping-runs.feature [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    Given Tertestrial is running a long-running test
+    When sending the command:
+      """
+      {"stopCurrentTest": true}
+      """
+    And sending the command:
+      """
+      {"stopCurrentTest": true}
+      """
+    Then I see "You have already killed bin/run-long-test"
```
Comment (charlierudolph) [inline features/stopping-runs.feature]: Do you think its worth having three different error messages? I think a generic: "no running test to stop" would be sufficient for each case.

File: src/command-runner.ls [hunk 3]
```diff
@@ -135,12 +139,15 @@ class CommandRunner
...
 
 
-  _stop-running-test: (done) ->
-    | !@current-process             =>  return done!
-    | @current-process?.exit-code?  =>  return done!
-    | @current-process?.killed      =>  return done!
+  _stop-running-test: (warn, done) ->
+    command = @_get-template(@current-command) if @current-command
+    switch
+    | !@current-process             =>  warn and console.log 'No test run so far' ; return done?!
```
Comment (charlierudolph) [inline src/command-runner.ls]: Thoughts on using the `error` helper that is used elsewhere in this file instead of `console.log`

File: src/command-runner.ls [hunk 4]
```diff
@@ -135,12 +139,15 @@ class CommandRunner
...
+    | @current-process?.exit-code?  =>  warn and console.log "#{command} has finished already" ; return done?!
+    | @current-process?.killed      =>  warn and console.log "You have already killed #{command}" ; return done?!
+    console.log bold "stopping #{command}"
     @current-process
-      ..on 'exit', -> done!
+      ..on 'exit', -> done?!
```
Comment (charlierudolph) [inline src/command-runner.ls]: Why was is this change needed? It doesn't look like you introduced a call without done.

**Comment by charlierudolph** [comment]
@kevgo My last round of comments weren't addressed. Any thoughts on them?

File: features/stopping-runs.feature (hunk 2 above)
Comment (kevgo) [inline features/stopping-runs.feature]: Why not be specific if we can be? I think meaningful and detailed feedback is part of a good user experience. In my opinion we just got used to generic error messages from our poorly written tools, and now expect that behavior. It doesn't have to be that way.

File: src/command-runner.ls (hunk 3 above)
Comment (kevgo) [inline src/command-runner.ls]: Good idea. Initially I thought this is just a warning, but printing this as an error message is fine here.

File: src/command-runner.ls (hunk 4 above)
Comment (kevgo) [inline src/command-runner.ls]: When the PipeListener receives a command via the named pipe, it emits a `command-received` event. The core has set the method `CommandRunner.run-command` as the handler for this event (in https://github.com/Originate/tertestrial-server/blob/master/src/index.ls#L58). 
That method calls `CommandRunner._stop-command` in https://github.com/Originate/tertestrial-server/blob/master/src/command-runner.ls#L50.

When called as an event listener, it is called without a callback. Which should be fine, since there is no reply that the terrestrial server should send back to the editor when done executing a command.

I agree this is a bit smelly. I don't want to add callback just because, though.

**Comment by kevgo** [comment]
@charlierudolph sorry I accidentally shipped this as part of shipping all the other PRs that you approved. I have implemented your open comments in #83.
